plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,562 @@
+        interface_group_policy_path = "/l3outTemplate/l3outs/{0}/interfaceGroups/{1}".format(l3out_object.index, match.index if match else "-")
+
+    ops = []
+
+    attributes_dict = {"bfd": bfd, "bfdMultiHop": bfd_multi_hop, "ospf": ospf}
+    empty_interface_routing_policy = False
+    empty_custom_qos_policy = False
+
+    if state == "present":
+        check_authentication_requirements(mso, attributes_dict)
+        mso_values = dict(
+            name=name,
+            description=description,
+            interfaceRoutingPolicyRef=interface_routing_policy_uuid,
+            qosPriority=qos_priority,
+            qosRef=custom_qos_policy_uuid,
+        )
+        if interface_routing_policy:
+            empty_interface_routing_policy = check_if_all_elements_are_none(list(interface_routing_policy.values()))
+            if not empty_interface_routing_policy:
+                mso_values["interfaceRoutingPolicyRef"] = mso_template.get_l3out_interface_routing_policy_uuid(
+                    mso, interface_routing_policy.get("name"), interface_routing_policy.get("template")
+                )
+        if custom_qos_policy:
+            empty_custom_qos_policy = check_if_all_elements_are_none(list(custom_qos_policy.values()))
+            if not empty_custom_qos_policy:
+                mso_values["qosRef"] = mso_template.get_custom_qos_policy_uuid(mso, custom_qos_policy.get("name"), custom_qos_policy.get("template"))
+
+        if match:
+            mso_values_remove = list()
+
+            if (interface_routing_policy_uuid == "" or empty_interface_routing_policy) and match.details.get("interfaceRoutingPolicyRef"):
+                mso_values_remove.append("interfaceRoutingPolicyRef")
+            if (custom_qos_policy_uuid == "" or empty_custom_qos_policy) and match.details.get("qosRef"):
+                mso_values_remove.append("qosRef")
+
+            for attribute_name, attribute_dict in attributes_dict.items():
+                if attribute_dict is not None:
+                    attribute_is_empty = check_if_all_elements_are_none(list(attribute_dict.values()))
+
+                    if match.details.get(attribute_name, {}).get("key", {}).get("ref"):
+                        match.details[attribute_name]["key"].pop("ref", None)
+                        mso.previous[attribute_name]["key"].pop("ref", None)
+                        mso.existing[attribute_name]["key"].pop("ref", None)
+
+                    if attribute_is_empty and match.details.get(attribute_name):
+                        mso_values_remove.append(attribute_name)
+
+                    elif not attribute_is_empty and not match.details.get(attribute_name):
+                        update_mso_values_attributes_payload(mso_values, {attribute_name: attribute_dict})
+
+                    elif not attribute_is_empty and match.details.get(attribute_name):
+                        mso_values[(attribute_name, "enabled")] = attribute_dict.get("enable")
```

Comment:
Why are we sometimes calling update_mso_values_attributes_payload ( also for non existing ) and we have other logic below. Is the creation of mso_values for these protocol with authentication settings not always equal?

With this comment I do not mean the removal of config.